src/main/java/org/jenkinsci/plugins/workflow/flow/StashManager.java

```diff
@@ -147,18 +162,18 @@ public static void unstash(@Nonnull Run<?,?> build, @Nonnull String name, @Nonnu
      * @throws AbortException in case there is no such saved stash
      * @see StashAwareArtifactManager#unstash
      */
-    public static void unstash(@Nonnull Run<?,?> build, @Nonnull String name, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull EnvVars env, @Nonnull TaskListener listener) throws IOException, InterruptedException {
+    public static List<File> unstash(@Nonnull Run<?,?> build, @Nonnull String name, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull EnvVars env, @Nonnull TaskListener listener) throws IOException, InterruptedException {
         Jenkins.checkGoodName(name);
         StashAwareArtifactManager saam = stashAwareArtifactManager(build);
         if (saam != null) {
             saam.unstash(name, workspace, launcher, env, listener);
-            return;
+            return Collections.emptyList();
```

Comment:
So this is clearly wrong: will not work with `artifact-manager-s3`.

Reply:
artefact managers like s3 and azure will be done later.  Besides, what should be done with your own jcloud artefact manager?

Reply:
That is `artifact-manager-s3`, which includes a generic layer based on JClouds plus an implementation using the JClouds binding for S3. Read https://jenkins.io/jep/202 before touching anything in this area.